### PR TITLE
Add test_spec for FCM

### PR DESCRIPTION
**Pull request by paulb777**
- Add unit test spec for Messaging.
- Add a complete Messaging build and test rule.
- Testing is done via the `pod gen` generated workspace.
- Merge iOS 8 test fixes from @maksymmalyhin's #3094
- Fix build warnings in test source.
- The old build environment is still supported.
- It will be removed when all pods have transitioned to test_specs and pod gen.
- For now, see the instructions at https://github.com/firebase/firebase-ios-sdk/blob/master/Functions/README.md.
- Use `--local-sources=./` with `pod gen` to include local InstanceID in workspace: 
- ` pod gen FirebaseMessaging.podspec --local-sources=./ --auto-open`

### CODE REVIEW

**Comment by paulb777** [comment]
Note that travis failures are from the unrelated #3022

File: Example/Messaging/Tests/FIRMessagingRemoteNotificationsProxyTest.m [hunk 1]
```diff
@@ -181,26 +176,26 @@ - (void)tearDown {
 
 - (void)testSwizzlingNonAppDelegate {
   RandomObject *invalidAppDelegate = [[RandomObject alloc] init];
-  [OCMStub([self.mockSharedApplication delegate]) andReturn:invalidAppDelegate];
+  [[GULAppDelegateSwizzler sharedApplication] setDelegate:(id<UIApplicationDelegate>)invalidAppDelegate];
```
Comment (ryanwilson) [inline Example/Messaging/Tests/FIRMessagingRemoteNotificationsProxyTest.m]: Any concerns here about using a singleton for these tests?

Comment (paulb777) [reply Example/Messaging/Tests/FIRMessagingRemoteNotificationsProxyTest.m]: Maybe, @maksymmalyhin can answer for sure. However, this fixes the tests being completely broken on iOS 8. See #3094 and #3093

Comment (paulb777) [reply Example/Messaging/Tests/FIRMessagingRemoteNotificationsProxyTest.m]: Specifically, see https://github.com/firebase/firebase-ios-sdk/blob/3ab4e388648c6f441717e8cbc68155dd5df7e335/GoogleUtilities/Example/Tests/Swizzler/GULAppDelegateSwizzlerTest.m#L645

Comment (ryanwilson) [reply Example/Messaging/Tests/FIRMessagingRemoteNotificationsProxyTest.m]: SGTM, let's keep an eye out for flakes just in case.

**Pull request merged**